scripts/eurostat/gdp/import_data.py

```diff
@@ -0,0 +1,150 @@
+        # "PPS_HAB_EU27_2020": "Purchasing power standard (PPS, EU27 from 2020), per inhabitant in \
+        #                       percentage of the EU27 (from 2020) average"
+    }
+    DESIRED_COLUMNS = ['geo', 'time'] + list(UNIT_CODES.keys())
+
+    # TODO (fpernice): Add a human-readable description of the codes to statVarObs
+    REGION_CODES = dict(json.loads(open('region_codes.json').read()))
+
+    def __init__(self):
+        self.raw_df = None
+        self.preprocessed_df = None
+        self.clean_df = None
+        print(self.REGION_CODES['EU27_2020'])
+
+    def download_data(self):
+        """Downloads raw data from Eurostat website and stores it in instance
+        data frame.
+        """
+        self.raw_df = pd.read_table(self.DATA_LINK)
+
+    def preprocess_data(self):
+        """Preprocesses instance raw_df and puts it into long format."""
+        if self.raw_df is None:
+            raise ValueError("Uninitialized value of raw data frame. Please "
+                             "check you are calling download_data before "
+                             "preprocess_data.")
+        self.preprocessed_df = preprocess_df(self.raw_df)
+
+    def clean_data(self):
+        """Drops unnecessary columns that are not needed for data import."""
+        if self.raw_df is None:
+            raise ValueError("Uninitialized value of processed data frame. "
+                             "Please check you are calling preprocess_data "
+                             "before clean_data.")
+        self.clean_df = self.preprocessed_df[self.DESIRED_COLUMNS]
+        self.clean_df = self.clean_df.replace(to_replace=':', value='')
+        new_col_names = {}
+        one_million = 1000000
+
+        def float_converter(val):
+            try:
+                return float(val)
+            except ValueError:
+                return float('nan')
+
+        for col in self.DESIRED_COLUMNS:
+            if col not in ['geo', 'time']:
+                self.clean_df[col] = self.clean_df[col].apply(float_converter)
+            if "MIO" in col:
+                new_col_names[col] = col.replace("MIO", "NUM")
+                self.clean_df[col] *= one_million
+            else:
+                new_col_names[col] = col
+        self.clean_df = self.clean_df.rename(new_col_names, axis=1)
+
+    def save_csv(self, filename='eurostat_gdp.csv'):
+        """Saves instance data frame to specified CSV file.
+
+        Raises:
+            ValueError: The instance clean_df data frame has not been
+            initialized. This is probably caused by not having called
+            process_data.
+        """
+        if self.clean_df is None:
+            raise ValueError("Uninitialized value of clean data frame. Please "
+                             "check you are calling clean_data before "
+                             "save_csv.")
+        self.clean_df.to_csv(filename)
+
+    def generate_tmcf(self):
+        temp = ('Node: E:eurostat_gdp->E{i}\n'\
```

Comment:
Do we need the ending slashes if it is within parentheses?

Reply:
Removed.